app/assets/javascripts/preferences/panes/two-factor-auth/TwoFactorAuth.ts

```diff
@@ -0,0 +1,120 @@
+import { action, makeAutoObservable, observable } from 'mobx';
+import { MfaGateway } from './MfaProps';
+import { TwoFactorActivation } from './TwoFactorActivation';
+
+type TwoFactorStatus =
+  | 'two-factor-enabled'
+  | TwoFactorActivation
+  | 'two-factor-disabled';
+
+export const is2FADisabled = (s: TwoFactorStatus): s is 'two-factor-disabled' =>
+  s === 'two-factor-disabled';
+
+export const is2FAActivation = (s: TwoFactorStatus): s is TwoFactorActivation =>
+  (s as any).type === 'two-factor-activation';
+
+export const is2FAEnabled = (s: TwoFactorStatus): s is 'two-factor-enabled' =>
+  s === 'two-factor-enabled';
+
+export class TwoFactorAuth {
+  private _status: TwoFactorStatus | 'fetching' = 'fetching';
+  private _errorMessage: string | null;
+
+  constructor(private readonly mfa: MfaGateway) {
```

Comment:
The mfa property when read and used throughout this code really doesn't add any context as to what exactly the object is. MFA stands for "multi-factor authenticator" so to name an object that doesn't make too much sense.

```suggestion
  constructor(private readonly gateway: MfaGateway) {
```

Reply:
Called it `mfaGateway`. Resolving comment.